backend/api/Areas/Leases/Controllers/SecurityDepositReturnController.cs

```diff
@@ -99,17 +103,23 @@ public IActionResult UpdateDepositReturn(long leaseId, long depositId, ParentCon
         /// Delete the specified return deposit on the passed lease.
         /// </summary>
         /// <returns></returns>
-        [HttpDelete("{leaseId:long}/deposit-returns")]
+        [HttpDelete("{leaseId:long}/deposits/{depositId:long}/returns")]
         [HasPermission(Permissions.LeaseEdit)]
         [Produces("application/json")]
         [ProducesResponseType(typeof(IEnumerable<LeaseModel>), 200)]
         [SwaggerOperation(Tags = new[] { "lease" })]
-        public IActionResult DeleteDepositReturn(long leaseId, [FromBody] ParentConcurrencyGuardModel<SecurityDepositReturnModel> deleteRequest)
+        public IActionResult DeleteDepositReturn(long leaseId, long depositId, [FromBody] ParentConcurrencyGuardModel<SecurityDepositReturnModel> deleteRequest)
         {
             if (leaseId != deleteRequest.ParentId)
             {
                 throw new BadRequestException($"Concurrency parent id mismatch.");
             }
+
+            if (depositId != deleteRequest.Payload.ParentDepositId)
+            {
+                throw new BadRequestException($"Bad payload id.");
```

Comment:
Just as an FYI, these messages are exposed to users on the frontend via the bomb icon.